### PR TITLE
Change the storage path for local/offline drafts

### LINKED ISSUE #3337: review path of storage of local offline drafts
This is after #3310 lands

I would like to review this in another issue and another PR - I'd rather have a more structured place where to save drafts.

For example a "localDrafts" array or object where all the drafts would be.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/pull/3310#discussion_r559498021_

### PR DESCRIPTION
**Pull request by limonte**
This PR puts all local drafts into 1 place in storage, under the `local_drafts` key. Also, it adds the migration, so users will not lose their local drafts which were saved in the old format.

close #3337

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/background_page/migrations.ts [hunk 1]
```diff
@@ -38,13 +40,27 @@ const addKeyInfoFingerprints = async () => {
...
+  const storageLocal = await storageLocalGetAll();
+  if (typeof globalStore.local_drafts === 'undefined') {
+    globalStore.local_drafts = {};
+  }
+  for (const key of Object.keys(storageLocal)) {
+    if (key.startsWith('local-draft-')) {
+      console.log(key);
+      console.info(`migrating local draft ${key}`);
+      globalStore.local_drafts[key] = storageLocal[key] as GmailRes.GmailDraftGet;
+      await storageLocalRemove([key]);
```
Comment (rrrooommmaaa) [inline extension/js/background_page/migrations.ts:60]: Perhaps, it'd be better to remove old drafts after successful saving to GlobalStore?

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]

File: extension/js/background_page/migrations.ts (hunk 1 above)
Comment (limonte) [reply extension/js/background_page/migrations.ts]: Agree, done 👍

File: extension/js/background_page/migrations.ts [hunk 2]
```diff
@@ -38,13 +40,30 @@ const addKeyInfoFingerprints = async () => {
...
+  const globalStore = await GlobalStore.get(['key_info_store_fingerprints_added', 'local_drafts']);
   if (!globalStore.key_info_store_fingerprints_added) {
     console.info('migrating KeyStorage to add fingerprints and emails of each key...');
     await addKeyInfoFingerprints();
     await GlobalStore.set({ key_info_store_fingerprints_added: true });
     console.info('done migrating');
   }
+  // migrate local drafts (#3337)
+  const storageLocal = await storageLocalGetAll();
+  if (typeof globalStore.local_drafts === 'undefined') {
```
Comment (rrrooommmaaa) [inline extension/js/background_page/migrations.ts:52]: Perhaps, if `globalStore.local_drafts` is not undefined, we should not call `storageLocalGetAll()`
(as with other migrations, we can consider this migration done and never return to it).
Or, perhaps, we could just load a flag like `local_drafts_moved` to not cause excessive load here?
Will there rarely be more than one draft?

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]

File: extension/js/background_page/migrations.ts (hunk 2 above)
Comment (limonte) [reply extension/js/background_page/migrations.ts:51]: > Perhaps, if globalStore.local_drafts is not undefined, we should not call storageLocalGetAll()
(as with other migrations, we can consider this migration done and never return to it).

I went with this suggestion, thank you 👍

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**